### PR TITLE
ci: run tests with and without PAPI debug enabled

### PR DESCRIPTION
**Pull request by gcongiu**
## Pull Request Description
Tests should make sure real use cases work as expected. Some tests might not working correctly if -O0 is used as optimization level in the compiler. For example, the ROCm runtime submits a kernel of 4 waves if the tests are built using -O0, which makes the tests fail.

## Author Checklist
- [x] **Description**
_Why_ this PR exists. Reference all relevant information, including _background_, _issues_, _test failures_, etc
- [x] **Commits**
_Commits_ are self contained and only do one thing
_Commits_ have a header of the form: `module: short description`
_Commits_ have a body (whenever relevant) containing a detailed description of the addressed problem and its solution
- [x] **Tests**
The PR needs to pass all the tests

### CODE REVIEW

**Comment by gcongiu** [comment]
@G-Ragghianti the idea with this PR is to have PAPI tested with debug both on and off. I am not sure the way I have done things is the right though. Could you provide any feedback here? Thanks

**Comment by G-Ragghianti** [comment]
I recommend against doing two builds within the same action because you need to make sure that the builds don't interfere with each other, when one fails, it will not run the other, and when there is a failure it isn't immediately clear which build caused the failure.  I recommend that we modify main.yaml to include a new axis "debug" in the actions matrix (with values "no" and "yes").  Then the existing ci.sh can be modified to reference this variable in the configure step.

File: .github/workflows/ci.sh [hunk 1]
```diff
@@ -43,9 +44,9 @@ if [ "$COMPONENT" = "infiniband_umad" ]; then
 fi
 
 if [ "$COMPONENT" = "perf_event" ]; then
-  ./configure --with-debug=yes --enable-warnings
+  ./configure --with-debug=$DEBUG --enable-warnings
```
Comment (gcongiu) [inline .github/workflows/ci.sh:47]: @G-Ragghianti will this work?

Comment (G-Ragghianti) [reply .github/workflows/ci.sh:47]: Yes I think that will work.

**Review by G-Ragghianti — APPROVED** [review]

**Pull request merged**